src/client.c

```diff
@@ -1678,25 +1678,47 @@ pretty_results (gchar* run_dir)
         g_free (std_err);
 }
 
-gchar *
+static gchar **
 parse_host (const gchar *connect_uri)
 {
-    gchar *host = NULL;
-    GRegex *regex;
-    GMatchInfo *match_info;
+    GMatchInfo  *match_info;
+    GRegex      *regex;
+    gchar      **ssh_destination;
+
+    g_return_val_if_fail (connect_uri != NULL, NULL);
+
+    /* [user@]hostname[:port][/ssh_port] */
+    regex = g_regex_new ("^(([\\w\\.\\-]+)@)?([\\w\\.\\-]+)((:\\d+)?(/\\d+)?)?$", 0, 0, NULL);
```

Comment:
We are missing proper escape in `(/\\d+)?`
First character is not escaped

Reply:
> First character is not escaped 

The forward slash? It doesn't need escape here.

Reply:
Martin:  If you look at his example, / is handling the / preceding ssh_port.